agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/MainEntryPoint.java

```diff
@@ -64,6 +71,8 @@ public static void start(Instrumentation instrumentation, URL bootstrapURL) {
             Configuration config = ConfigurationBuilder.create(agentJarFile.toPath());
             configuration = config.instrumentationSettings;
             configPath = config.configPath;
+            BasicFileAttributes attributes = Files.readAttributes(configPath, BasicFileAttributes.class);
+            lastModifiedTime = attributes.lastModifiedTime().toMillis();
```

Comment:
let's capture `lastModifiedTime` before reading the file, just to completely eliminate race condition of the file being updated in between reading it and capturing the `lastModifiedTime`